controllers/k8ssandra/remove_dc_test.go

```diff
@@ -142,6 +145,16 @@ func deleteDcWithStargateAndReaper(ctx context.Context, t *testing.T, f *framewo
 		},
 	}
 
+	require.Eventually(func() bool {
```

Comment:
Do we need Eventually blocks here and below? No other controller is running in this test than k8ssandra controller, so the update of the dc status should never fail, right?

Reply:
I've tried removing them, and the test fails. There's a delay between the patching/creation of the kc objects and their processing by the controller, where we can end up with the test code going too fast. It's also why I added a check on whether or not the generations are different before updating the observed generation.